src/server/pfs/s3/s3_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+// Tests for the PFS' S3 emulation API. Note that, in calls to
+// `tu.UniqueString`, all lowercase characters are used, unlike in other
+// tests. This is in order to generate repo names that are also valid bucket
+// names. Otherwise minio complains that the bucket name is not valid.
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	minio "github.com/minio/minio-go"
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+	"github.com/pachyderm/pachyderm/src/server/pfs/server"
+	tu "github.com/pachyderm/pachyderm/src/server/pkg/testutil"
+)
+
+func serve(t *testing.T, pc *client.APIClient) (*http.Server, uint16) {
+	port := tu.UniquePort()
+	srv := Server(pc, port)
+
+	go func() {
+		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
+			t.Fatalf("http server returned an error: %v", err)
+		}
+	}()
+
+	// Wait for the server to start
+	c := &http.Client{}
+	for i := 0; i < 50; i++ {
+		res, err := c.Get(fmt.Sprintf("http://127.0.0.1:%d/_ping", port))
+		if err == nil && res.StatusCode == 200 {
+			return srv, port
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
```

Comment:
It's a little more standard in our tests to use a `backoff.Retry` loop for stuff like this than a busy wait. See an example [here](https://github.com/pachyderm/pachyderm/blob/master/src/server/pachyderm_test.go#L559).